charts/tidb-operator/templates/controller-manager-deployment.yaml

```diff
@@ -1,7 +1,8 @@
+{{- if .Values.controllerManager.create }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: tidb-controller-manager
+  name: tidb-controller-manager-{{ .Release.Name }}
```

Comment:
There are compatibility issues here when upgrade operator.

Reply:
Yes, when use `helm upgrade` the old release, deployment `tidb-controller-manager` will be recreate as `tidb-controller-manager-{{ .Release.Name }}`

Reply:
@shonge @mikechengwei to keep backward compatibility, what about adding `appendReleaseSuffix: false` to the values.yaml and if set it to `true`, we use the name here.

Reply:
@shonge @mikechengwei What about add a field `appedReleaseSuffix: false` to the values.yaml of the tidb-operator chart:

- If set to true, append the deployment name of controller-manager, the service account of controller-manager, and scheduler with the release name suffix, so that
    - If users install another operator release in the same namespaces as the current one, they have to set `appedReleaseSuffix: true`, no need to change anything else, and the two releases can be upgraded/deleted independently
    - If users install another operator release in another namespace (this should be the recommended way in our doc), no need to change anything in the values.yaml and the two releases can be upgraded/deleted independently
- If set to false or not set, keep the current behavior.